tools/js-sdk-release-tools/packages/typescript-codegen-breaking-change-detector/src/azure/diff/declaration-diff.ts

```diff
@@ -383,6 +387,129 @@ export function findTypeAliasBreakingChanges(source: TypeAliasDeclaration, targe
   return [createDiffPair(DiffLocation.TypeAlias, DiffReasons.TypeChanged, sourceNameNode, targetNameNode)];
 }
 
+export function findClassDeclarationBreakingChanges(source: ClassDeclaration, target: ClassDeclaration): DiffPair[] {
+  const targetConstructors = target.getConstructors();
+  const sourceConstructors = source.getConstructors();
+  const constructorBreakingChanges = findConstructorBreakingChanges(sourceConstructors, targetConstructors);
+
+  const targetProperties = target.getType().getProperties();
+  const sourceProperties = source.getType().getProperties();
+  const propertyBreakingChanges = findPropertyBreakingChanges(sourceProperties, targetProperties);
+
+  //source.getMembers().forEach((p) => console.log(p.getText() + ' ' + p.getCombinedModifierFlags()));
+  const targetMembers = target.getMembers();
+  const sourceMembers = source.getMembers();
+  const memberBreakingChanges = findMemberBreakingChanges(sourceMembers, targetMembers);
+  
+  return [...constructorBreakingChanges, ...propertyBreakingChanges, ...memberBreakingChanges];
+}
+
+export function isSameConstructor(left: ConstructorDeclaration, right: ConstructorDeclaration): boolean {
+  const leftOverloads = left.getOverloads()
+  const rightOverloads = right.getOverloads()
+  const overloads = leftOverloads.filter((t) => {
+    const compatibleSourceFunction = rightOverloads.find((s) => {
+      // NOTE: isTypeAssignableTo does not work for overloads
+      const parameterPairs = [
+        ...findParameterBreakingChangesCore(s.getParameters(), t.getParameters(), '', '', s, t),
+        ...findParameterBreakingChangesCore(t.getParameters(), s.getParameters(), '', '', t, s),
+      ];
+      return parameterPairs.length === 0;
+    });
+    return compatibleSourceFunction === undefined;
+  });
+  return overloads.length === 0;
+}
+
+
+function findConstructorBreakingChanges(
```

Comment:
try to combine with `findCallSignatureBreakingChanges`, making similar functions with slightly is hard to mantain